samples/bench_initend.c

```diff
@@ -0,0 +1,58 @@
+/* Benchmark {de,in}flateInit and {de,in}flateEnd */
+#include <assert.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <string.h>
+
+#include "nxu.h"
+#include "zlib.h"
+
+uint64_t tb_freq = 0; /* needed by nxu.h to cache clock frequency */
+
+static inline double timebase_average_us(uint64_t sum, int count) {
+        double avg = (double)sum / count;
+
+        return avg * 1000000 / nx_get_freq();
```

Comment:
User code should not have access to `nx_get_freq()`.

That being said, I don't think it should be used in a program that is under "samples" because that isn't a good example on how users should write programs using the NX API.

Reply:
@tuliom Do you think the same applies to `nx_get_time` and `nx_time_diff`?

At first I had created a separate `bench` directory just for this. But later I realized that `make bench` actually compiles everything under samples. I believe some of the files under `samples` are actually better classified as micro-benchmarks and could also go to that new dir, but this was out-of-scope for this PR, so I left it under `samples`.

Reply:
> Do you think the same applies to nx_get_time and nx_time_diff?

@mscastanho yes.
I also agree that reworking the samples directory is out-of-scope.

Reply:
I changed the test to stop using the nx_* timebase-related functions from nxu.h.